async-query/src/main/java/org/opensearch/sql/spark/scheduler/parser/IntervalScheduleParser.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.sql.spark.scheduler.parser;
+
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import org.opensearch.jobscheduler.spi.schedule.IntervalSchedule;
+
+/** Parse string raw schedule into job scheduler IntervalSchedule */
+public class IntervalScheduleParser {
+
+  private static final Pattern DURATION_PATTERN =
+      Pattern.compile(
+          "^(\\d+)\\s*(years?|months?|weeks?|days?|hours?|minutes?|minute|mins?|seconds?|secs?|milliseconds?|millis?|micros?|nanoseconds?|nanos?)$",
+          Pattern.CASE_INSENSITIVE);
+
+  public static IntervalSchedule parse(String intervalStr, Instant startTime) {
```

Comment:
Is it a right decision to implement it by ourselves? Is there common format for time duration and library for that?

Reply:
> Is it a right decision to implement it by ourselves? Is there common format for time duration and library for that?

The problem here is to converting Spark CalendarInterval to OpenSearch job scheduler interval.

Reply:
Why do we use Spark CalendarInterval?

Reply:
Spark CalendarInterval is the interface for spark scheduler, we will need this when convert stream job to external scheduler